vcd/resource_vcd_multisite_org_association.go

```diff
@@ -0,0 +1,234 @@
+	// The existence of an XML file containing an associated Org is just a convenient way of retrieving the Org ID.
+	associationData, err := org.GetOrgAssociationByOrgId(associatedOrgId)
+	if err != nil {
+		return diag.Errorf("association data not found for Org '%s' with org ID '%s': %s", org.AdminOrg.Name, associatedOrgId,
+			fmt.Errorf("%s: %s", err, govcd.ErrorEntityNotFound))
+	}
+	dSet(d, "associated_org_id", associatedOrgId)
+	dSet(d, "associated_org_name", associationData.OrgName)
+	dSet(d, "associated_site_id", associationData.SiteID)
+	dSet(d, "status", associationData.Status)
+	d.SetId(associatedOrgId)
+
+	return nil
+}
+
+// resourceVcdOrgAssociationUpdate only deals with optional check of association status, as there is nothing else that can be updated
+func resourceVcdOrgAssociationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*VCDClient)
+	orgId := d.Get("org_id").(string)
+	associatedOrgId := d.Id()
+	org, err := client.GetAdminOrgById(orgId)
+	if err != nil {
+		return diag.Errorf("error retrieving Org '%s': %s", orgId, err)
+	}
+	connectionCheckMinutes := d.Get("connection_timeout_mins").(int)
+	if d.HasChange("connection_timeout_mins") && connectionCheckMinutes > 0 {
+		status, elapsed, err := org.CheckOrgAssociation(associatedOrgId, time.Minute*time.Duration(connectionCheckMinutes))
+		if err != nil {
+			return diag.Errorf("error checking for org connection after %s - detected status '%s': %s", elapsed, status, err)
+		}
+	}
+	return resourceVcdOrgAssociationRead(ctx, d, meta)
+}
+
+func resourceVcdOrgAssociationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*VCDClient)
+	orgId := d.Get("org_id").(string)
+	org, err := client.GetAdminOrgById(orgId)
+	if err != nil {
+		return diag.Errorf("error retrieving Org '%s': %s", orgId, err)
+	}
+	associatedOrgId := d.Id()
+	if associatedOrgId == "" {
+		associatedOrgId = d.Get("associated_org_id").(string)
+	}
+	if associatedOrgId == "" {
+		return diag.Errorf("no site ID found in either d.Id() or 'associated_org_id' field")
+	}
+	associationData, err := org.GetOrgAssociationByOrgId(associatedOrgId)
+	if err != nil {
+		return diag.Errorf("association data not found for Org '%s' with org ID '%s': %s", org.AdminOrg.Name, associatedOrgId, err)
+	}
+	err = org.RemoveOrgAssociation(associationData.Href)
+	if err != nil {
+		return diag.Errorf("error removing association data for Org '%s' to org '%s': %s", org.AdminOrg.Name, associationData.OrgName, err)
+	}
+	return nil
+}
+
+func resourceVcdOrgAssociationImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client := meta.(*VCDClient)
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 2 {
+		return nil, fmt.Errorf("[org association import] org association must be provided as local-org-id.remote-org-id")
```

Comment:
I think this syntax is missing in the docs

Reply:
Added